src/main/java/org/jitsi/jicofo/ParticipantChannelAllocator.java

```diff
@@ -233,6 +203,70 @@ else if (reInvite)
         }
     }
 
+    /**
+     * Creates and sends a Jingle IQ (either {@code session-accept} or
+     * {@code transport-replace}) and sends it to the {@code participant}.
+     * Blocks until a response is received or a timeout occurs.
+     *
+     * @param address the destination JID.
+     * @param contents the list of contents to include.
+     * @return {@code false} on failure.
+     * @throws OperationFailedException if we are unable to send a packet
+     * because the XMPP connection is broken.
+     */
+    private boolean sendJingleIq(
```

Comment:
it seems a little odd to me that we have this generic `sendJingleIq` function which has to then look at fields to figure out which JingleIq actually needs to be created and sent.  are there not other places (which presumably trigger/call this) which already know what is happening and can create the necessary more directly?

Reply:
> are there not other places (which presumably trigger/call this) which already know what is happening and can create the necessary more directly?

Yes. There's only one place where this is used, and the only reason I extracted it is to avoid having the code inline (it's 50 lines of code in a method that is already over 50 lines)

Reply:
ok i see now.  is the comment inaccurate, though?  it says "creates either `session-accept` or `transport-replace`" but the code creates either `session-initiate` or `transport-replace`.  and, while fixing that (unless I'm misunderstanding) could you add to the comment how it decides to make one vs the other?  (it looks like it's based on the `reInvite` member variable or the participant having a jingle session?)